File: 119.杨辉三角-ii.py
#
# @lc app=leetcode.cn id=119 lang=python3
#
# [119] 杨辉三角 II
#

# @lc code=start
from typing import List
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        res = [1]
        if rowIndex == 0:
            return res
        for i in range(1, int(rowIndex/2)+1):
            res.append(int(res[i-1]*(rowIndex-i+1)/i))
        if rowIndex%2 == 1:
            res = res + res[len(res)-1::-1]
        else:
            res = res + res[len(res)-2::-1]
        return res
if __name__ == "__main__":
    so = Solution()
    #print(so.getRow(4))

# @lc code=end

File: 119.杨辉三角-ii.java
/*
 * @lc app=leetcode.cn id=119 lang=java
 *
 * [119] 杨辉三角 II
 */

// @lc code=start
import java.util.List;
import java.util.ArrayList;

class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> nums = new ArrayList<>();
        List<Integer> nums_reverse = new ArrayList<>();
        nums.add(1); 
        nums_reverse.add(0, 1);
        if(rowIndex==0){
            return nums;
        }
        for(int i=1; i<rowIndex/2+1; i++){
            int a = (int)((long)nums.get(i-1)*(rowIndex-i+1)/i);
            nums.add(a);
            //System.out.println(a);
            nums_reverse.add(0, a);
        }
        if(rowIndex%2 == 0){
            nums_reverse.remove(0);
            nums.addAll(nums_reverse);
            return nums;
        }
        else{
            nums.addAll(nums_reverse);
            return nums;
        }
    }
}

// @lc code=end

